src/monster.cpp

```diff
@@ -80,84 +80,87 @@ const mtype_id mon_zombie_tough( "mon_zombie_tough" );
 
 monster::monster()
 {
- position.x = 20;
- position.y = 10;
- position.z = -500; // Some arbitrary number that will cause debugmsgs
- wandf = 0;
- hp = 60;
- moves = 0;
- def_chance = 0;
- friendly = 0;
- anger = 0;
- morale = 2;
- faction = mfaction_id( 0 );
- mission_id = -1;
- no_extra_death_drops = false;
- dead = false;
- made_footstep = false;
- unique_name = "";
- hallucination = false;
- ignoring = 0;
- upgrades = false;
- upgrade_time = -1;
+    position.x = 20;
+    position.y = 10;
+    position.z = -500; // Some arbitrary number that will cause debugmsgs
+    unset_dest();
+    wandf = 0;
+    hp = 60;
+    moves = 0;
+    def_chance = 0;
+    friendly = 0;
+    anger = 0;
+    morale = 2;
+    faction = mfaction_id( 0 );
+    mission_id = -1;
+    no_extra_death_drops = false;
+    dead = false;
+    made_footstep = false;
+    unique_name = "";
+    hallucination = false;
+    ignoring = 0;
+    upgrades = false;
+    upgrade_time = -1;
 }
 
 monster::monster( const mtype_id& id )
```

Comment:
Two of the constructors could probably be changed to use the third, either the one with no arguments or the one with two.